src/Evolution/Systems/ScalarWave/BoundaryCorrections/UpwindPenalty.hpp

```diff
@@ -0,0 +1,279 @@
+ * denote quantities on the internal and external side of the mortar. Note that
+ * Eq. (6.3) of \cite Teukolsky2015ega is not exactly what's implemented since
+ * that boundary term does not consistently treat both sides of the interface on
+ * the same footing.
+ *
+ * For the scalar wave system the correction is:
+ *
+ * \f{align}{
+ *   D_{\Psi} &= \tilde{\lambda}_{v^{\Psi}}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},\Psi}
+ *                - \tilde{\lambda}_{v^{\Psi}}^{\mathrm{int}}
+ *                v^{\mathrm{int},\Psi}, \\
+ *   D_{\Pi} &= \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{ext}}
+ *             v^{\mathrm{ext},+} +
+ *             \tilde{\lambda}_{v^-}^{\mathrm{ext}}
+ *             v^{\mathrm{ext},-}\right)
+ *             + \tilde{\lambda}_{v^\Psi}^\mathrm{ext}\gamma_2
+ *             v^{\mathrm{ext},\Psi}
+ *             \notag \\
+ *           &-\frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{int}}
+ *             v^{\mathrm{int},+} +
+ *             \tilde{\lambda}_{v^-}^{\mathrm{int}}
+ *             v^{\mathrm{int},-}\right)
+ *             - \tilde{\lambda}_{v^\Psi}^\mathrm{int}\gamma_2
+ *             v^{\mathrm{int},\Psi} , \\
+ *   D_{\Phi_{i}}
+ *              &= \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},+}
+ *                - \tilde{\lambda}_{v^-}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},-}\right)n_i^{\mathrm{ext}}
+ *                + \tilde{\lambda}_{v^0}^{\mathrm{ext}}
+ *                v^{\mathrm{ext},0}_{i}
+ *                \notag \\
+ *              &-
+ *                \frac{1}{2}\left(\tilde{\lambda}_{v^+}^{\mathrm{int}}
+ *                v^{\mathrm{int},+}
+ *                - \tilde{\lambda}_{v^-}^{\mathrm{int}}
+ *                v^{\mathrm{int},-}\right)n_i^{\mathrm{int}}
+ *                - \tilde{\lambda}_{v^0}^{\mathrm{int}}
+ *                v^{\mathrm{int},0}_{i},
+ * \f}
+ *
+ * with characteristic fields
+ *
+ * \f{align}{
+ *   v^{\Psi} &= \Psi, \\
+ *   v^{0}_{i} &= (\delta^k_i-n^k n_i)\Phi_{k}, \\
+ *   v^{\pm} &= \Pi\pm n^i\Phi_{i} -\gamma_2 \Psi,
+ * \f}
+ *
+ * and characteristic speeds
+ *
+ * \f{align}{
+ *   \lambda_{v^\Psi} =& -v^i_g n_i, \\
+ *   \lambda_{v^0} =& -v^i_g n_i, \\
+ *   \lambda_{v^\pm} =& \pm 1 - v^i_g n_i,
+ * \f}
+ *
+ * where \f$v_g^i\f$ is the mesh velocity. We have also defined
+ *
+ * \f{align}{
+ *   \tilde{\lambda}_{\hat{\alpha}} =
+ *    \left\{
+ *        \begin{array}{ll}
+ *          \lambda_{\hat{\alpha}} &
+ *            \mathrm{if}\;\lambda_{\hat{\alpha}}\le 0.0 \\
+ *          0 & \mathrm{otherwise}
+ *        \end{array}\right.
+ * \f}
+ *
+ * Note that we have assumed \f$n_i^{\mathrm{ext}}\f$ points in the same
+ * direction as \f$n_i^{\mathrm{int}}\f$. If \f$n_i^{\mathrm{ext}}\f$ points in
+ * the opposite direction the external speeds have their sign flipped and the
+ * \f$\pm\f$ fields and their speeds reverse roles. Specifically, in the code we
```

Comment:
I'm not sure I understand the `speeds reverse roles` with the formula as written -- it appears from the current form that the v^+ and v^- appear in the same way as the previous equation.

Reply:
I'm adding:
```cpp
  * \f$\pm\f$ fields and their speeds reverse roles (i.e. the                                                                                                                                                                                                                     
  * \f$v^{\mathrm{ext},+}\f$ field is now flowing into the element, while                                                                                                                                                                                                         
  * \f$v^{\mathrm{ext},-}\f$ flows out). Specifically, in the code we have:
```
Does that clarify things? I think the role reversal only matters in the `Phi_i` term because they aren't being added, and there it cancels out with the minus sign that the `n_i^ext` brings in